### PR TITLE
Fix broken handle server for Java 1.8 (DSpace 6.x version) - WIP

### PR DESCRIPTION
**Pull request by OyvindLGjesdal**
## References

* See #8557
* Related to Handle Server

## Description
Updates to dependencies for dspace-handle server for the `dspace-6_x` branch, currently fails during build, with mismatch when pulling in dependencies. Handle server not tested for 6_x branch.

## Instructions for Reviewers

List of changes in this PR:
Based on DSpace 7 handle changes

* Update handle dependency from dspace artifact to handle.net (add the maven handle repo) and upgrade version to 9.3.0.
* Change `Streamtable` implementation to `net.cnri.util.StreamTable` (DS7 PR)
* Update and change to packages `bcpkix-jdk15on`, `bcprov-jdk15on` dependencies (from DS7 PRs) used by handle server.
* Add new dependencies `net.cnri:cnri-servlet-container` and `òrg.eclipse.jetty:jetty-server` for handle package (DS7 PRs)
* Exclusion of `jetty-server` and `cnri-servlet-container` from `dspace-api` dependency in *dspace-xmlui* for resolving dependency collision.
* Exclusion of `cnri-servlet-container` from `dspace-api` dependency in  `dspace-oai`, `dspace-rest` and `dspace-modules-additions`. 
* Exclusion of `asm-commons` and `asm` from `elasticsearch` dependency for resolving dependency collision in `dspace-api`.  
```
Warning:  Rule 0: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
Failed while enforcing releasability the error(s) are [
Dependency convergence error for org.ow2.asm:asm:7.0 paths to dependency are:
+-org.dspace:dspace-api:6.5-SNAPSHOT
  +-net.cnri:cnri-servlet-container:3.0.0
    +-org.eclipse.jetty:jetty-annotations:9.4.15.v20190215
      +-org.ow2.asm:asm:7.0
and
+-org.dspace:dspace-api:6.5-SNAPSHOT
  +-org.elasticsearch:elasticsearch:1.4.0
    +-org.ow2.asm:asm:4.1
``` 
* Exclusion of `javax.annotation-api` from `jersey-client` in `dspace-api`.
```
Dependency convergence error for javax.annotation:javax.annotation-api:1.3 paths to dependency are:
+-org.dspace:dspace-api:6.5-SNAPSHOT
  +-net.cnri:cnri-servlet-container:3.0.0
    +-org.eclipse.jetty:jetty-annotations:9.4.15.v20190215
      +-javax.annotation:javax.annotation-api:1.3
and
+-org.dspace:dspace-api:6.5-SNAPSHOT
  +-org.glassfish.jersey.core:jersey-client:2.22.1
    +-org.glassfish.jersey.core:jersey-common:2.22.1
      +-javax.annotation:javax.annotation-api:1.2
```

* Patch `make-handle-config` to be identical to version in 7.4.
* Update DummyHttpServletRequest.java test with boilerplate code generated from IntelliJ to add missing stubs after upgrade.
* Add a `gson.version` property and use same version for `dspace-xmlui` and `dspace-api`.

```
Warning:  
Dependency convergence error for com.google.code.gson:gson:2.8.1 paths to dependency are:
+-org.dspace:dspace-xmlui:6.5-SNAPSHOT
  +-org.dspace:dspace-api:6.5-SNAPSHOT
    +-gr.ekt.bte:bte-io:0.9.3.5
      +-com.google.code.gson:gson:2.8.1
and
+-org.dspace:dspace-xmlui:6.5-SNAPSHOT
  +-org.dspace:dspace-api:6.5-SNAPSHOT
    +-com.google.code.gson:gson:2.8.1
and
+-org.dspace:dspace-xmlui:6.5-SNAPSHOT
  +-com.google.code.gson:gson:2.6.1
```

**Include guidance for how to test or review your PR.** This may include: steps to reproduce a bug, screenshots or 
description of a new feature, or reasons behind specific changes. See failing job above.

* Run the `make-handle-config` script and insert dummy settings
* Start the `handle-server` script
* review the output of `handle-server/logs` error and access files.
* Especially test the components which exclude older versions to check if they still work.

** Some questions for @tdonohue 
* Should annotation-api be added as a dependency for 1.3, when cnri-servlet-container is excluded from multiple modules and annotation api is excluded from jersey-client, there `annotation-api` could not be present in the classpath at all?
* There is a big jump between the asm and asm-commons versions in elasticsearch and jetty-annotations, so they are probably not compatible. I tried to figure out what the dependencies are used for, and I think it is for generating bytecode for the lucene expression language. Is this used by the module or could it be excluded and ignored?
* Since the handle server and dependencies run as a standalone application, I assume that the colliding dependencies can be excluded from the different modules. Does this hold?
* Should `bcmail-jdk15` also be updated or removed?

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [ ] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [ ] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [ ] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [ ] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Comment by OyvindLGjesdal** [comment]
I'll edit the list of changes in the PR and comments tomorrow but at least the CI completes now.

Maybe publishing a fixed old handle-server is the better option.

I.e some of the upgrades could break java 7 support (like for dspace 5) (Jetty lists 8 as supported for the latest 9.x version).

**Comment by OyvindLGjesdal** [comment]
What do you think @tdonohue? This is as far as I can get myself with this PR I think.

**Comment by tdonohue** [comment]
@OyvindLGjesdal : Thanks for your work here!  I've been on holiday for the last week & busy with workshops just before that...so I haven't been able to look at this PR for almost a month now.  Skimming it now, I do see you've asked a lot of questions in the description & the implication of several is that this upgrade to Handle v9 _may not be as easy as it seems_ (and may have unintended side effects).   I haven't yet had time to do a more thorough review, but will see if I can get back to this later this week.

In the meantime, I'm linking this PR up with your original ticket (#8557) to make it more easily findable to others...currently it isn't mentioned in that ticket.

File: dspace/bin/make-handle-config [hunk 1]
```diff
@@ -46,23 +53,24 @@ echo "n" >>$tempfile  # Do not encrypt admin private key
...
-$BINDIR/dspace dsrun net.handle.server.SimpleSetup $handledir < $tempfile >/dev/null
+$BINDIR/dspace dsrun net.handle.server.SimpleSetup $handledir < $tempfile >/dev/null 2>&1
 
 # Remove temp file
 rm -f $tempfile
 
 # Now update server configuration (config.dct) with our handle prefix and the
 # DSpace Handle plugin
-sed 's/YOUR_NAMING_AUTHORITY/'$handle'/' <$handledir/config.dct >$tempfile
+sed 's/YOUR_PREFIX/'$handle'/' <$handledir/config.dct >$tempfile
```
Comment (misilot) [inline dspace/bin/make-handle-config:63]: I think keeping it `YOUR_NAMING_AUTHORITY` makes sense, since that matches the install documentation for 6x?

Comment (ilittle-cnri) [reply dspace/bin/make-handle-config:63]: That line is updating the default Handle config file, which uses `YOUR_PREFIX` in Handle 9. So, that change is correct, but means the DSpace 6 docs won't match anymore.

**Comment by ilittle-cnri** [comment]
I've started looking this over. It looks basically correct (haven't tried running it yet), but the big version jumps and the `YOUR_NAMING_AUTHORITY` vs `YOUR_PREFIX` thing are setting off warnings in my the back of my mind that upgrading DSpace 6 to use Handle 9 might not be the correct path.

I'll continue poking it and see if anything breaks. I suspect, though, that having folks use a modified Handle 6.2 jar would be an easier and safer solution.

**Comment by tdonohue** [comment]
@ilittle-cnri : FWIW, if there *is* a modified Handle 6.2 JAR that people on DSpace 6.x could continue to use, I **agree that'd be the best solution here.**  However, to my knowledge no such Handle 6.2 JAR exists which is compatible with the latest JDK 8 (per the issue described in #8557).  If CNRI (or someone else) is able to create such a JAR, then that'd be quite simple to swap DSpace 6 over to using it (as it's just changing the dependency in the POM)

**Comment by ilittle-cnri** [comment]
I'll talk to folks to see if that's something we can do, or at least assist with. Would it be most helpful if it was in a Maven repository, or is just a link to a jar sufficient?

**Comment by tdonohue** [comment]
@ilittle-cnri : most helpful would definitely be if it was in a Maven Repository, as that means DSpace just needs to update our POMs.  However, if that's not possible, then a JAR is second best... but, please be aware that may mean we'd have to release that JAR to our own DSpace maven repo just to make it easier for DSpace sites to use/install.  Thanks!

**Comment by mwoodiupui** [comment]
Someone has created such a JAR (6.2 repaired for recent JRE), and it's working here.  Read on down through #8557 to see the thread by ebalnaves.

**Pull request closed without merging**